app/models/manageiq/providers/ansible_tower/automation_manager/machine_credential.rb

```diff
@@ -1,2 +1,66 @@
 class ManageIQ::Providers::AnsibleTower::AutomationManager::MachineCredential < ManageIQ::Providers::AutomationManager::Authentication
+  FIELDS = {
+    :description => {
+      :type       => :string,
+      :required   => false,
+      :label      => _('Description'),
+      :help_text  => _('Optional description of this credential.'),
+      :encrypted  => false
+    },
+    :username         => {
+      :type       => :string,
+      :required   => false,
+      :label      => _('Username'),
+      :help_text  => _('Username for this credential.'),
+      :max_length => 1024,
+      :encrypted  => false
+    },
```

Comment:
(I've renamed `FIELDS` => `EXTRA_ATTRIBUTES`). So this `EXTRA_ATTRIBUTES` will serve as hint to UI what inputs it will gather from user. What will be stored in `authentications.options` is controlled by the return from Tower API, which will essentially *nullify* the secrets. 

There is a minor issue. UI would still need to have the knowledge of mapping field names.  e.g. Tower API's create/update credential expect the parameter hash key to be `username` (vs `authentications` using `userid`).  Refresh/inventory would need to map the other way.  The more overlapping fields between `authentications` and Tower parameters, the more knowledge the code will need. So another approach is to capture all Tower parameters (including `username`, `password`) in `EXTRA_ATTRIBUTES` which UI/refresh will blindly follow.